polynote-kernel/src/main/scala/polynote/kernel/util/TaskManager.scala

```diff
@@ -0,0 +1,14 @@
+package polynote.kernel.util
+
+import cats.effect.IO
+import cats.effect.concurrent.Deferred
+
+// TODO: trying to centralize how tasks are tracked.
```

Comment:
+1! Were you thinking of potentially being able to see the list of tasks that have run, etc.? That could be super-useful for debugging. 

I like the idea of thinking about what happens in Polynote as a sequence of `State => Transform(State): State`, in which case we have RuntimeSymbolTable to keep track of the different States and TaskManager to keep track of the different Transforms. 

Of course it's more complicated than this because RST only keeps track of symbol state and not UI state...

Reply:
Or I guess as we discussed, a Tree of `State => Transform(State): State` 😉